### PR TITLE
Add intel_amx backend for Radix Attention for CPU

### PR DESCRIPTION
**Pull request by yanbing-j**
## Motivation

This PR is a follow-up on https://github.com/sgl-project/sglang/issues/2807 to add intel_amx attention backend for CPU. Users are suggested to only set device to cpu. If device is set to cpu and attention backend is not set, then it will use intel_amx attention backend as default. And if hardware doesn't supports it, it will fallback to torch_native.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 1]
```diff
@@ -340,7 +340,7 @@ def init_new(
             ret.extend_prefix_lens = torch.tensor(
                 batch.extend_prefix_lens, dtype=torch.int32
             ).to(device, non_blocking=True)
-            if model_runner.server_args.attention_backend != "torch_native":
+            if support_triton(model_runner.server_args.attention_backend):
```
Comment (mingfeima) [inline python/sglang/srt/model_executor/forward_batch_info.py:354]: @yanbing-j we are very likely to provide an optimized implementation e.g.  `compute_position_cpu` when the cpu kernels are compiled.

File: python/sglang/srt/utils.py [hunk 2]
```diff
@@ -2126,3 +2126,21 @@ def load_json_config(data: str):
 
 def dispose_tensor(x: torch.Tensor):
     x.set_(torch.empty((0,), device=x.device, dtype=x.dtype))
+
+
+def support_triton(backend: str):
+    if backend in ["torch_native", "intel_amx"]:
+        return False
+    return True
```
Comment (mingfeima) [inline python/sglang/srt/utils.py]:
```suggestion
def support_triton(backend: str) -> bool:
    return backend not in ["torch_native", "intel_amx"]
```

**Review by mingfeima — APPROVED** [review]
generally ok, just some minor issues to address.

**Review by mingfeima — CHANGES_REQUESTED** [review]
Add a log when intel amx backend is used, to clarify the actual use case.

**Review by mingfeima — APPROVED** [review]

**Comment by yanbing-j** [comment]
Hi @zhyncs, could you please help review this PR? The CI failures are not related to CPU. Thanks!

File: sgl-kernel/python/sgl_kernel/cpu.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+def fused_experts(
+    x,
+    w13_weight,
+    w2_weight,
+    topk_weights,
+    topk_ids,
+    inplace,
+    is_vnni=True,
+):
+    return sgl_kernel.common_ops.fused_experts_cpu(
```
Comment (zhyncs) [inline sgl-kernel/python/sgl_kernel/cpu.py]: Remove `sgl_kernel.common_ops`

Comment (zhyncs) [reply sgl-kernel/python/sgl_kernel/cpu.py]: The others are the same.

**Review by zhyncs — CHANGES_REQUESTED** [review]

**Comment by zhyncs** [comment]
Hi @mingfeima @yanbing-j Before landing this PR, can we add the sgl-kernel-cpu workflow and publish it to PyPI?

**Comment by zhyncs** [comment]
ref https://github.com/sgl-project/sglang/pull/6452#issuecomment-2903848337 All unit tests passed in the latest main.

It would be beneficial to include this build and test in the CI workflow. @mingfeima @yanbing-j

File: sgl-kernel/python/sgl_kernel/cpu.py
Comment (Alcanderian) [inline sgl-kernel/python/sgl_kernel/cpu.py:1]: I think this file is redundant.

**Comment by ZailiWang** [comment]
> ref [#6452 (comment)](https://github.com/sgl-project/sglang/pull/6452#issuecomment-2903848337) All unit tests passed in the latest main.
> 
> It would be beneficial to include this build and test in the CI workflow. @mingfeima @yanbing-j

Yeah, we are working on this.

File: sgl-kernel/python/sgl_kernel/cpu.py
Comment (yanbing-j) [reply sgl-kernel/python/sgl_kernel/cpu.py:1]: Have removed `cpu.py`, use `torch.ops.sgl_kernel.func_cpu` directly.

File: sgl-kernel/python/sgl_kernel/cpu.py (hunk 3 above)
Comment (yanbing-j) [reply sgl-kernel/python/sgl_kernel/cpu.py]: Have removed `cpu.py`, use `torch.ops.sgl_kernel.func_cpu` directly.

File: python/sglang/srt/layers/attention/base_attn_backend.py [hunk 4]
```diff
@@ -109,3 +109,7 @@ def forward_extend(
     ):
         """Run a forward for extend."""
         raise NotImplementedError()
+
+    def support_triton(self):
```
Comment (zhyncs) [inline python/sglang/srt/layers/attention/base_attn_backend.py:113]: QQ why do we need to add `support_triton`?

Comment (yanbing-j) [reply python/sglang/srt/layers/attention/base_attn_backend.py:113]: The detailed information is in https://github.com/sgl-project/sglang/pull/6143#discussion_r2094345182.

one advantage of doing this is that for future hardwares, ascend npu, ppu, etc. they only need to instantiate support_triton when they implement radix attention and don't have to modify schedule_batch.py.

**Review by zhyncs — APPROVED** [review]
